src/runtime/tracing.cpp

```diff
@@ -149,9 +149,12 @@ WEAK int32_t default_trace(void *user_context, const halide_trace_event *e) {
                         ss << ((uint64_t *)(e->value))[i];
                     }
                 } else if (e->type_code == 2) {
-                    halide_assert(user_context, print_bits >= 32 && "Tracing a bad type");
+                    halide_assert(user_context, print_bits >= 16 && "Tracing a bad type");
                     if (print_bits == 32) {
                         ss << ((float *)(e->value))[i];
+                    } else if (print_bits == 16) {
+                        // TODO: We should also emit a decimal approximation
+                        ss.write_float16_from_bits_as_hex_float( ((uint16_t *)(e->value))[i] );
```

Comment:
This is intended to be a human-readable number. I doubt many humans can read hex floats. I'd cast to double then print instead. If you're tracing and you want the precise value then you're using the binary format anyway, not this code.